src/Components/CardList/index.jsx

```diff
@@ -1,21 +1,28 @@
-import React, { useContext } from 'react';
+import React, {useEffect} from 'react';
 import Card from './Card';
-import { CardContext } from '../../Context/card-context';
+import { useSelector, useDispatch } from 'react-redux';
+import { onSave } from '../../Store/Actions/actions';
+import { fetchCards } from '../../Store/Actions/cards';
 
-const CardList = () => {
-  const { cards, view, onSave } = useContext(CardContext);
+const CardList = props => {
+  const { cards, view } = useSelector(state => (state));
+  const dispatch = useDispatch();
+  const onSaveChanges = id => { dispatch(onSave(cards, id)) };
+  useEffect(() => {
+    if (!cards.length) dispatch(fetchCards());
+  })
 
   return (
     <div className='main'>
-      {cards.map(card => {
+      {cards && cards.map(card => {
         return (
           <Card
-            key={card.id}
-            card={card}
-            onView={view}
-            onSaveChanges={onSave} />
-        )
-      })}
+          key={card.id}
+          card={card}
+          onView={view}
+          onSaveChanges={(val) => onSaveChanges(val)}
+          route={props} />
```

Comment:
Passing all the component props into a nested component - bad idea, there could be many redundant ones. Pass only the ones that are really required.

Reply:
Fixed